src/containers/modals/ledger/LedgerConnect.js

```diff
@@ -31,25 +33,60 @@ import networkService from 'services/networkService';
 
 import * as styles from './LedgerConnect.module.scss';
 
+const steps = {
+  usingLedger: 'USING_LEDGER',
+  selectAddress: 'SELECT_ADDRESS'
+};
+
 function LedgerConnect ({ submit, open }) {
   const dispatch = useDispatch();
   const [ loading, setLoading ] = useState(false);
+  const [ getConnectedAddressLoading, setGetConnectedAddressLoading ] = useState(false);
+  const [ contractDataError, setContractDataError ] = useState(false);
+
+  const [ step, setStep ] = useState(steps.usingLedger);
+
+  const [ selectedAddress, setAddress ] = useState('');
+  const [ selectedPath, setPath ] = useState('');
+
+  useEffect(() => {
+    async function fetchConnectedAddress () {
+      setGetConnectedAddressLoading(true);
+      try {
+        const { path, address } = await networkService.getConnectedLedgerAddress();
+        setAddress(address);
+        setPath(path);
+        setGetConnectedAddressLoading(false);
+      } catch (error) {
+        setGetConnectedAddressLoading(false);
```

Comment:
can we move this and line 59 out of `try-catch` since it does the same thing for both cases?